packages/app/src/components/search/use-techdocs-entities.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { stringifyEntityRef } from '@backstage/catalog-model';
+import { useApi } from '@backstage/core-plugin-api';
+import {
+  CATALOG_FILTER_EXISTS,
+  catalogApiRef,
+  entityPresentationApiRef,
+} from '@backstage/plugin-catalog-react';
+import useAsync from 'react-use/esm/useAsync';
+
+export function useTechdocsEntities() {
+  const catalogApi = useApi(catalogApiRef);
+  const presentationApi = useApi(entityPresentationApiRef);
+
+  const { value } = useAsync(async () => {
+    // Return a list of entities which are documented.
+    const { items } = await catalogApi.getEntities({
+      fields: ['metadata.name', 'metadata.namespace', 'kind'],
+      filter: {
+        'metadata.annotations.backstage.io/techdocs-ref': CATALOG_FILTER_EXISTS,
+      },
+    });
+
+    const presentationEntities = await Promise.all(
+      items.map(
+        entity => presentationApi.forEntity(stringifyEntityRef(entity)).promise,
+      ),
+    );
+    const entityNameByRef = new Map(
```

Comment:
cc @awanlin, I think we should roll back the previous change, as using the presentation API this way doesn't scale. For example, if we receive 1000 entities, we'll end up sending approximately 1000 requests back to the server to retrieve essentially the same data we already have.

I propose rolling back [#76738ec4dd7270516a301e74e7cea01fd9775d49](https://github.com/backstage/backstage/pull/28782/commits/76738ec4dd7270516a301e74e7cea01fd9775d49).

Since we're discussing this, @grantila, do you think it would make sense to add a limit to `getEntities`, or would that introduce side effects?

Reply:
That's fine with me, thanks for catching this @vinzscam 👍 

Sorry for the extra work on this @grantila